components/kyma-environment-broker/internal/storage/postsql/generic_read_test.go

```diff
@@ -0,0 +1,81 @@
+package postsql_test
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/dbmodel"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage/postsql"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenericRead(t *testing.T) {
+
+	ctx := context.Background()
+
+	t.Run("Should Read Instance by Id", func(t *testing.T) {
+		containerCleanupFunc, cfg, err := storage.InitTestDBContainer(t.Logf, ctx, "test_DB_1")
+		require.NoError(t, err)
+		defer containerCleanupFunc()
+
+		cleanupNetwork, err := storage.SetupTestNetworkForDB(ctx)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer cleanupNetwork()
+
+		tablesCleanupFunc, err := storage.InitTestDBTablesWithPath(t, cfg.ConnectionURL(), "./../../../../schema-migrator/migrations/kyma-environment-broker/")
+		require.NoError(t, err)
+		defer tablesCleanupFunc()
+
+		logger := logrus.StandardLogger()
+
+		connection, err := postsql.InitializeDatabase(cfg.ConnectionURL(), 10, logger)
+		assert.NoError(t, err)
+
+		defer connection.Close()
+
+		instanceId := "instance_id"
+
+		session := connection.NewSession(nil)
+		defer session.Close()
+
+		tx, err := session.Begin()
+		assert.NoError(t, err)
+
+		testServiceName := "Test Service Name"
+
+		_, err = session.InsertInto(postsql.InstancesTableName).
```

Comment:
in my opinion, the PR should be about Read and Write operation so in the unit test we can test "write" and "read" operation then compare objects if are equals